apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,522 @@
+
+		fd = open(g_file[0], O_WROK | O_CREAT);
+		if (fd < 0) {
+			printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+
+		ret = write(fd, g_rw_buf, g_fsz);
+		if (ret != (g_fsz)) {
+			printf("Unable to write in file: %s, size: %d, ret: %d\n", g_file[0], g_fsz, ret);
+			ret = ERROR;
+			goto close_with_fd;
+		}
+
+		close(fd);
+
+		if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+			fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+			ret = -1;
+			goto close_with_frt;
+		}
+
+		tot_time  = after.timeleft - before.timeleft;
+		printf("Total time taken is: %lu us\n", tot_time);
+
+		ret = unlink(g_file[0]);
+		if (ret != OK) {
+			printf("Failed to unlink file %s, errno : %d\n", g_file[0], errno);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+		break;
+
+	case TEST_FLAG_OVERWRITE :
+		printf("\n****************************************************************************************\n");
+		printf("**********************************TEST OVERWRITE************************************");
+		printf("\n****************************************************************************************\n");
+		printf("File size: %d bytes, Buffer size: %d bytes\n", g_fsz, g_bsz);
+
+		fd = open(g_file[0], O_WROK | O_CREAT);
+		if (fd < 0) {
+			printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+
+		for (i = 0; i < (g_fsz / g_bsz); i++) {
+			ret = write(fd, g_rw_buf, g_bsz);
+			if (ret != (g_bsz)) {
+				printf("Unable to write in file: %s, ret: %d\n", g_file[0], ret);
+				ret = ERROR;
+				goto close_with_fd;
+			}
+		}
+		close(fd);
+
+		for (i = 1; i <= ITR_OVE; i++) {
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			fd = open(g_file[0], O_WROK);
+			if (fd < 0) {
+				printf("Unable to open file : %s, fd: %d\n", g_file[0], fd);
+				ret = ERROR;
+				goto close_with_frt;
+			}
+
+			ret = lseek(fd, 0, SEEK_SET);
+			if (ret != 0) {
+				printf("Unable to move file pointer to start of the file\n");
+				ret = ERROR;
+				goto close_with_fd;
+			}
+
+			for (j = 1; j <= (g_fsz / g_bsz); j++) {
+				ret = write(fd, g_rw_buf, g_bsz);
+				if (ret != (g_bsz)) {
+					printf("Unable to write in file: %s, iter= %d ,ret = %d\n", g_file[0], j, ret);
+					ret = ERROR;
+					goto close_with_fd;
+				}
+			}
+
+			close(fd);
+
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+				fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+				ret = -1;
+				goto close_with_frt;
+			}
+
+			itr_time = after.timeleft - before.timeleft;
+			tot_time += itr_time;
+			printf("iteration: %d, time: %lu\n", i, itr_time);
+		}
+		printf("total time: %lu us\n", tot_time);
+
+		ret = unlink(g_file[0]);
```

Comment:
Can unlinks also be clubbed for whatever cases possible at the end of this function?

Reply:
Since unlinks are in loop and each TC has different requirement for unlink. Like in GC effect it is intended to induce gc and likwise.